java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1953,6 +1955,54 @@ public void updateSystemInfo(JsonElement jsonResult, String minionId) {
         });
     }
 
+    public boolean updateMaster(MinionServer minion, String master) {
+        boolean changed = false;
+
+        Optional<Server> proxy = ServerFactory.lookupProxyServer(master);
+        if (proxy.isPresent()) {
+                // the system is connected to a proxy
+                // check if serverPath already exists
+                if (! ServerFactory.findServerPath(minion, proxy.get()).isPresent()) {
+                    // proxy path does not exist -> create it
+                    Set<ServerPath> proxyPaths = ServerFactory.createServerPaths(minion, proxy.get(), master);
+                    minion.getServerPaths().clear();
+                    minion.getServerPaths().addAll(proxyPaths);  //.orElse(Collections.emptySet()));
+
+                    ServerFactory.save(minion);
+                    changed = true;
+                }
+         }
+         else {
+                if (! minion.getServerPaths().isEmpty()) {
+                    // reconnecting from proxy to master
+                    minion.getServerPaths().clear();
+
+                    ServerFactory.save(minion);
+                    changed = true;
+                }
+        }
+        if (changed) {
+            // Regenerate the pillar data
+            MinionPillarManager.INSTANCE.generatePillar(minion);
+
+            // push the changed pillar data to the minion
+            saltApi.refreshPillar(new MinionList(minion.getMinionId()));
+
+                ApplyStatesAction action = ActionManager.scheduleApplyStates(minion.getCreator(),
```

Comment:
Just to help me understand the need to update channels.
If the master to where the minion was connected changed, then the channels/repository URLs should be updated at the minion to point to the right master. Is this the right reason?

Reply:
Yes, it updates `/etc/zypp/repos.d/susemanager:channels.repo` on the minion